test/functional/rpc_createmultisig.py

```diff
@@ -133,117 +84,165 @@ def check_addmultisigaddress_errors(self):
-        elif self.output_type == 'p2sh-segwit':
+        elif output_type == 'p2sh-segwit':
             desc = 'sh(wsh({}))'.format(desc)
-        elif self.output_type == 'bech32':
+        elif output_type == 'bech32':
             desc = 'wsh({})'.format(desc)
         desc = descsum_create(desc)
 
-        msig = node2.createmultisig(self.nsigs, self.pub, self.output_type)
+        msig = node2.createmultisig(nsigs, pub_keys, output_type)
         assert 'warnings' not in msig
         madd = msig["address"]
         mredeem = msig["redeemScript"]
         assert_equal(desc, msig['descriptor'])
-        if self.output_type == 'bech32':
+        if output_type == 'bech32':
             assert madd[0:4] == "bcrt"  # actually a bech32 address
 
-        if self.is_bdb_compiled():
+        if wallet_multi is not None:
             # compare against addmultisigaddress
-            msigw = wmulti.addmultisigaddress(self.nsigs, self.pub, None, self.output_type)
+            msigw = wallet_multi.addmultisigaddress(nsigs, pub_keys, None, output_type)
             maddw = msigw["address"]
             mredeemw = msigw["redeemScript"]
             assert_equal(desc, drop_origins(msigw['descriptor']))
             # addmultisigiaddress and createmultisig work the same
             assert maddw == madd
             assert mredeemw == mredeem
-            wmulti.unloadwallet()
 
         spk = address_to_scriptpubkey(madd)
-        txid = self.wallet.send_to(from_node=self.nodes[0], scriptPubKey=spk, amount=1300)["txid"]
-        tx = node0.getrawtransaction(txid, True)
-        vout = [v["n"] for v in tx["vout"] if madd == v["scriptPubKey"]["address"]]
-        assert len(vout) == 1
-        vout = vout[0]
-        scriptPubKey = tx["vout"][vout]["scriptPubKey"]["hex"]
-        value = tx["vout"][vout]["value"]
-        prevtxs = [{"txid": txid, "vout": vout, "scriptPubKey": scriptPubKey, "redeemScript": mredeem, "amount": value}]
+        value = decimal.Decimal("0.00004000")
+        tx = self.wallet.send_to(from_node=self.nodes[0], scriptPubKey=spk, amount=int(value * COIN))
+        prevtxs = [{"txid": tx["txid"], "vout": tx["sent_vout"], "scriptPubKey": spk.hex(), "redeemScript": mredeem, "amount": value}]
 
         self.generate(node0, 1)
 
-        outval = value - decimal.Decimal("0.00001000")
-        rawtx = node2.createrawtransaction([{"txid": txid, "vout": vout}], [{self.final: outval}])
+        outval = value - decimal.Decimal("0.00002000")  # deduce fee (must be higher than the min relay fee)
+        # send coins to node2 when wallet is enabled
+        node2_balance = node2.getbalances()['mine']['trusted'] if self.is_wallet_compiled() else 0
+        out_addr = node2.getnewaddress() if self.is_wallet_compiled() else getnewdestination('bech32')[2]
+        rawtx = node2.createrawtransaction([{"txid": tx["txid"], "vout": tx["sent_vout"]}], [{out_addr: outval}])
 
         prevtx_err = dict(prevtxs[0])
```

Comment:
Again older code, but `prevtx_err` leads to an impression that it's an error related to a previous transaction but the usage of this variable doesn't show that it's an error, but a transaction itself.

Reply:
> Again older code, but `prevtx_err` leads to an impression that it's an error related to a previous transaction but the usage of this variable doesn't show that it's an error, but a transaction itself.

Sure. Will leave it as is to not continue expanding the diff. Feel free to tackle it on a follow-up.